src/lib.rs

```diff
@@ -44,6 +44,8 @@ pub struct Contract {
     pub authors: UnorderedMap<AccountId, HashSet<PostId>>,
     pub communities: UnorderedMap<CommunityHandle, Community>,
     pub featured_communities: Vec<FeaturedCommunity>,
+    pub available_addons: UnorderedMap<CommunityAddOnId, CommunityAddOn>,
+    pub addon_configurations: LookupMap<CommunityAddOnConfigId, CommunityAddOnConfig>,
```

Comment:
As mentioned above, let's move these to the Community struct. It does not make a lot of sense to reuse addon configurations between communities as it won't be possible to identify who has permission to edit the configurations.

Reply:
I don't think the intention is to reuse addon configurations between communities. The reason for a lookup map is to not having to load the config for all addons in a community. 

But when you mention that, what probably should be done is to make sure that a `communityAddonId` and `communityAddonConfigId` is unique to the community.

Reply:
I don't expect configs to be bigger than 1kb each and not more than 10 addons per community, so we are talking about 10kb. It is nothing from CPU time, calling a host function to read a single key is more expensive than deserializing 10kb. This "optimization" complicates things, and potentially makes it worse in the end.

Reply:
If you still want to go with storing CommunityAddOnConfig as individual keys, move `LookupMap` into `Community` struct and use the `StorageKey::CommunityAddonConfig(CommunityHandle)` as the storage key prefix.

Reply:
On the second thought, given that everything is already implemented, I vote for keeping it as is and address the collision issue - you can use a tuple as a key: `LookupMap<(CommunityHandle, CommunityAddOnConfigId), CommunityAddOnConfig>`